pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,162 @@
+	} else {
+		status.EndWithSuccess("Globalnet is disabled")
+	}
+
+	return true
+}
+
+func checkClusterGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	clusterGlobalEgress, err := cluster.SubmClient.SubmarinerV1().ClusterGlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining the clusterGlobalEgressIPs cr: %v", err))
+		return
+	}
+
+	if len(clusterGlobalEgress.Items) != 1 {
+		status.QueueFailureMessage(fmt.Sprintf("Number (%d) of clusterGlobalEgress resources != 1", len(clusterGlobalEgress.Items)))
+		return
+	}
+
+	if clusterGlobalEgress.Items[0].Name != constants.ClusterGlobalEgressIPName {
+		status.QueueFailureMessage(fmt.Sprintf("clusterGlobalEgress name (%s) not equals to default name", clusterGlobalEgress.Items[0].Name))
+		return
+	}
+}
+
+func checkGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	globalEgressIps, err := cluster.SubmClient.SubmarinerV1().GlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining GlobalEgressIPs resources: %v", err))
+		return
+	}
+
+	for i := range globalEgressIps.Items {
+		if *globalEgressIps.Items[i].Spec.NumberOfIPs != len(globalEgressIps.Items[i].Status.AllocatedIPs) {
+			status.QueueFailureMessage(fmt.Sprintf("Error GlobalEgressIPs(%s) NumberOfIPs != AllocatedIPs", globalEgressIps.Items[i].Name))
+			return
+		}
+	}
+}
+
+func checkGlobalIngressIps(cluster *cmd.Cluster, status *cli.Status) {
+	mcsClient, err := mcsClientset.NewForConfig(cluster.Config)
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining mcs client: %v", err))
+		return
+	}
+
+	serviceExports, err := mcsClient.MulticlusterV1alpha1().ServiceExports(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining ServiceExport resources: %v", err))
+		return
+	}
+
+	for i := range serviceExports.Items {
+		ns := serviceExports.Items[i].GetNamespace()
+		name := serviceExports.Items[i].GetName()
+
+		svc, err := cluster.KubeClient.CoreV1().Services(ns).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			status.QueueFailureMessage(fmt.Sprintf("Error matching Service resource for exported service(%s,%s).%v", ns, name, err))
+			return
+		}
+
+		if svc.Spec.Type == corev1.ServiceTypeClusterIP {
+			globalIngress, err := cluster.SubmClient.SubmarinerV1().GlobalIngressIPs(ns).Get(context.TODO(), name, metav1.GetOptions{})
+			if err != nil {
+				status.QueueFailureMessage(fmt.Sprintf("Error no GlobalIngressIP resource allocated for service(%s,%s).%v", ns, name, err))
+				return
+			}
+
+			if globalIngress.Status.AllocatedIP == "" {
+				status.QueueFailureMessage(fmt.Sprintf("Error GlobalIngressIP(%s,%s) - AllocatedIP is empty", ns, name))
+				return
```

Comment:
`AllocatedIP` **shouldn't** be empty if the status condition indicates success but doesn't hurt to verify.

```suggestion
				status.Failure("No global IP was allocated for the GlobalIngressIP associated with exported service \"%s/%s\"", ns, name)
				failed = true
				continue
```